src/Confluent.Kafka/Structs/Error.cs

```diff
@@ -64,8 +72,19 @@ public override int GetHashCode()
         public static bool operator !=(Error a, Error b)
             => !(a == b);
 
+        // Returns the string representation of the error.
+        // If a rich error string is available return that, otherwise fall
+        // back to librdkafka's static error code to string conversion.
         public override string ToString()
-            => $"[{(int)Code}] {Message}";
+        {
+            if (!string.IsNullOrEmpty(String))
+                return String;
```

Comment:
Okay, can we make a later pass when we have code formatter or lint checker in place?